src/Reflection/ReflectionProperty.php

```diff
@@ -84,6 +155,14 @@ public static function createFromInstance(object $instance, string $propertyName
         return $property;
     }
 
+    public static function cloneWithImplementingClass(self $property, ReflectionClass $implementingClass): self
```

Comment:
Could be `withImplementingClass()`.

Should probably be `@internal`

Reply:
Fixed.